### PR TITLE
workflows: add commitlint workflow

### PR DESCRIPTION
**Pull request by obnoxxx**
component definitions are basically taken from the existing git log.

Signed-off-by: Michael Adam <obnox@redhat.com>

### CODE REVIEW

**Comment by obnoxxx** [comment]
strange, I don't see why the new workflow file does not trigger a check run ... 🤷‍♂️

**Comment by obnoxxx** [comment]
> strange, I don't see why the new workflow file does not trigger a check run ... 🤷‍♂️

now it starts and even succeeds! 😄

File: .commitlintrc.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        "server",
+        "ad-server",
+        "tests",
+        "images",
+        "docs",
+        "examples",
+        "workflows",
+        "kubernetes",
+        "hack",
+	"Makefile"
```
Comment (synarete) [inline .commitlintrc.json:19]: We user `makefile:` (lowercase) labels so far. Better keep it that way.

File: .github/workflows/commitlint.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Commitlint
```
Comment (synarete) [inline .github/workflows/commitlint.yml:1]: Unlike `rook`, in `samba-operator` we use single workflow-action yaml, which allows us to see all jobs at once (as DAG) and halt workflow as soon a a job fails. Better have this section part of `main.yaml` with proper `needs`.

File: .github/workflows/commitlint.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      - master
+      - release-*
+  pull_request:
+    branches:
+      - master
+      - release-*
+
+jobs:
+  lint:
+    runs-on: ubuntu-18.04
```
Comment (synarete) [inline .github/workflows/commitlint.yml:16]: Use `ubuntu-latest`

File: .github/workflows/commitlint.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
```
Comment (synarete) [inline .github/workflows/commitlint.yml:20]: Better use latest `actions/checkout@v3`

File: .github/workflows/commitlint.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        with:
+          fetch-depth: 0
```
Comment (synarete) [inline .github/workflows/commitlint.yml:22]: Why do you need this? AFAIK, it is not needed in this case.

File: .github/workflows/commitlint.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      - uses: wagoid/commitlint-github-action@v2.0.3
```
Comment (synarete) [inline .github/workflows/commitlint.yml:23]: Use latest version: `wagoid/commitlint-github-action@v5`

**Review by synarete — COMMENTED** [review]
The type-enum looks ok, but see other comments in this PR

**Comment by obnoxxx** [comment]
converted to draft becaus @phlogistonjohn suggested we solve the corresponding effort for samba-operator first : https://github.com/samba-in-kubernetes/samba-operator/pull/272 or https://github.com/samba-in-kubernetes/samba-operator/pull/274

**Comment by anoopcs9** [comment]
Support for gitlint is already merged via #124.

**Pull request closed without merging**